src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py

```diff
@@ -36,12 +37,78 @@
 # pylint: disable=too-many-lines
 
 
+class MultiAPIAdaptor(object):
+    # We will bridge all the code difference here caused by SDK breaking change
+    def __init__(self, cli_ctx):
+        self.old_api = cli_ctx.cloud.profile == '2017-03-09-profile'
```

Comment:
There are going to be three profiles now. The word `old_api` may not be precise.